core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java

```diff
@@ -52,9 +54,32 @@ public static AuthRepository getInstance() {
 	}
 
 	@Override
-	public void setJWT(@NonNull String jwt) throws ClientException {
+	public void setJWT(@NonNull String jwt) throws ClientException, UserLoggedInException {
 		this.jwt = jwt;
-		localData.setJWT(jwt);
+		final JwtBody jwtBody = getJwtBody(jwt);
+		final String currentUserID = localData.getUserID();
+		localData.setJWT(jwtBody);
```

Comment:
if throw new UserLoggedInException don't call localData.setJWT(jwtBody);
The error tells the Host app to first log out and only then login again

Reply:
Also, we should call remote logout - we enforce the host app to do logout before login in with different user

Reply:
it's an internal exception not related to the DS.
so `Kin.java` will know there is already logged in user. and clear other cached data.